src/user/userRepository.ts

```diff
@@ -1,41 +1,67 @@
 import {UserData} from 'src/entities/user.type';
 import {Injectable} from '@nestjs/common';
 import {SearchUsersDto} from './dto/searchUsers.dto';
+import {SearchOneUser} from './dto/searchOneUser.dto';
+import {InjectRepository} from '@nestjs/typeorm';
+import {Repository, In} from 'typeorm';
+import {UserEntity} from 'src/db/user.entity';
 
 @Injectable()
 export class UserRepository {
-    constructor() {
-        //this.dbProvider = dbProvider;
-        //this.cacheProvider = cacheProvider;
-    }
+    constructor(
+        @InjectRepository(UserEntity)
+        private userRepository: Repository<UserEntity>,
+    ) {}
 
-    async findById(id: string): Promise<string> {
-        return id;
+    async findById(id: string): Promise<UserData | null> {
+        return this.userRepository.findOneBy({id});
     }
 
-    async findOne({ids, login, phone}: SearchUsersDto): Promise<object> {
-        return {ids, login, phone};
+    async findOne({id, login, phone}: SearchOneUser): Promise<UserData | null> {
+        return this.userRepository.findOneBy({id, login, phone});
```

Comment:
do i need to add something like `if (!result) {
            return null;
        }`